packages/coil-extension/src/background/services/Stream.ts

```diff
@@ -331,7 +331,8 @@ class StreamAttempt {
 
     return new Promise((resolve, reject) => {
       const onMoney = (sentAmount: string) => {
-        this.onMoney(sentAmount)
+        // Wait until `setImmediate` so that `connection.totalDelivered` has been updated.
```

Comment:
Nice work Mr @sentientwaffle !

Reply:
Maybe ilp-protocol-stream should be updated to set that beforehand ?

Reply:
Since updating the hold [may not change the amount](https://github.com/interledgerjs/ilp-protocol-stream/blob/762d4b36583ab70848b0f6d7fe69a4ac3d884644/src/stream.ts#L426) that isn't desirable.

Reply:
Can you elaborate on that? 
Note that I'm not a STREAM/ILP expert.

Reply:
You're right, stream could set the amount beforehand, but I think it would qualify as a breaking change (albeit just barely). I've made a note-to-self to include that change the next time I make a breaking stream change. I don't think it warrants one on its own since it's so simple to work around.

Reply:
Ok, so outgoing_money is when money is actually _delivered_ right? Or am I wrong there? 
It's definitely a change but one could reasonably argue it's actually a bug and the current behavior is incorrect. Would it really actually "break" any client code or simply give more correct results ?

Reply:
Yes, `outgoing_money` is the notification that money has been successfully delivered on a stream.